src/main/java/seedu/address/logic/parser/SearchCommandParser.java

```diff
@@ -0,0 +1,122 @@
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SearchCommand
+     * and returns a SearchCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SearchCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+            ArgumentTokenizer.tokenize(args, PREFIX_TID, PREFIX_NAME, PREFIX_PHONE, PREFIX_ADDRESS, PREFIX_COD,
+                PREFIX_DELIVERY_TIMESTAMP, PREFIX_WAREHOUSE, PREFIX_COMMENT, PREFIX_TYPE);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_TID, PREFIX_NAME, PREFIX_PHONE, PREFIX_ADDRESS, PREFIX_COD,
+            PREFIX_DELIVERY_TIMESTAMP, PREFIX_WAREHOUSE, PREFIX_COMMENT, PREFIX_TYPE)
+            && argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+        }
+
+        List<String> keywords;
+        if (!argMultimap.getPreamble().isEmpty()) {
+            keywords = Arrays.asList(argMultimap.getPreamble().split("\\s+"));
+            return new SearchCommand(new OrderContainsKeywordsPredicate(keywords));
+        }
+
+        KeywordContainsOrderPrefix keywordContainsOrderPrefix = new KeywordContainsOrderPrefix(argMultimap);
+        StringBuilder keywordsString = new StringBuilder();
+        if (keywordContainsOrderPrefix.getHasTid()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_TID);
+        }
+
+        if (keywordContainsOrderPrefix.getHasName()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_NAME);
+        }
+
+        if (keywordContainsOrderPrefix.getHasPhone()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_PHONE);
+        }
+
+        if (keywordContainsOrderPrefix.getHasAddress()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_ADDRESS);
+        }
+
+        if (keywordContainsOrderPrefix.getHasTimeStamp()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_DELIVERY_TIMESTAMP);
+        }
+
+        if (keywordContainsOrderPrefix.getHasWarehouse()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_WAREHOUSE);
+        }
+
+        if (keywordContainsOrderPrefix.getHasComment()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_COMMENT);
+        }
+
+        if (keywordContainsOrderPrefix.getHasCod()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_COD);
+        }
+
+        if (keywordContainsOrderPrefix.getHasItemType()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_TYPE);
+        }
```

Comment:
I think can extract this method out into another function, hopefully will solve the codacy error of NPath complexity 
`The method 'parse(String)' has an NPath complexity of 384`.